internal/provider/resource_gitlab_deploy_key_enable.go

```diff
@@ -127,17 +142,21 @@ func resourceGitlabDeployKeyEnableDelete(ctx context.Context, d *schema.Resource
 	return nil
 }
 
-func resourceGitlabDeployKeyEnableStateImporter(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
-	s := strings.Split(d.Id(), ":")
-	if len(s) != 2 {
-		d.SetId("")
-		return nil, fmt.Errorf("Invalid Deploy Key import format; expected '{project_id}:{deploy_key_id}'")
+func resourceGitLabDeployKeyEnableParseId(id string) (int, int, error) {
```

Comment:
The project id can actually be the `path_with_namespace` as string, too. Thus, this wouldn't work. Also the `d.Set("project", strconv.Itoa(project))` situations, won't work in those cases.

Reply:
I was unaware it was not converted to an int internally.  Do you have an example/suggestion of how to handle the case where a user would do an import with a `path_with_namespace ` and also I think the docs might need to be updated as well if that is allowable.  The import docs that I started working off of only has the projectID as an option for import.

Reply:
You don't need to do anything special. Just use a string - `go-gitlab` will do the correct thing for you.

And indeed, the documentation can be improved there and mention that either a project path or id can be used. (it's correctly documented for the `project` argument already though.)

Reply:
I updated the import documentation and I believe strings should work now for the project attribute